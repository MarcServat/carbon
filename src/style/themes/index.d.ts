import mintTheme from "./mint";
import aegeanTheme from "./aegean";
import sageTheme from "./sage";
import { ThemeObject } from "./base";

declare const carbonThemeList: ThemeObject[];

export { default as noTheme } from "./none";
export { default as baseTheme } from "./base";
export { mintTheme, aegeanTheme, sageTheme };
export { carbonThemeList };
