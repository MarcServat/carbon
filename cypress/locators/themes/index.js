import { getComponent } from "..";

export const theme = (themeName) => cy.get(`[data-theme="${themeName}"]`);
export const buttonToggleComponent = () =>
  getComponent("button-toggle").first().find("label");
export const linkComponent = () => getComponent("link");
export const loaderComponent = () => getComponent("loader").find("div");
