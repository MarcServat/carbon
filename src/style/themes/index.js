import mintTheme from "./mint";
import aegeanTheme from "./aegean";
import sageTheme from "./sage";

export { default as noTheme } from "./none";
export { default as baseTheme } from "./base";
export { mintTheme, aegeanTheme, sageTheme };
export const carbonThemeList = [mintTheme, aegeanTheme, sageTheme];
