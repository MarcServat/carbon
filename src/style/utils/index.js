import filterStyledSystemPaddingProps from "./filter-styled-system-padding-props";
import filterStyledSystemMarginProps from "./filter-styled-system-margin-props";

export { filterStyledSystemPaddingProps, filterStyledSystemMarginProps };
