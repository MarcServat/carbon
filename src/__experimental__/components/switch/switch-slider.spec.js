import React from "react";
import TestRenderer from "react-test-renderer";
import "jest-styled-components";
import { css, ThemeProvider } from "styled-components";
import {
  assertStyleMatch,
  carbonThemesJestTable,
} from "../../../__spec_helper__/test-utils";
import Loader from "../../../components/loader/loader.component";
import SwitchSlider from "./switch-slider.component";
import SwitchSliderPanel from "./switch-slider-panel.style";
import { baseTheme } from "../../../style/themes";

describe("SwitchSlider", () => {
  describe("base theme", () => {
    it("renders as expected", () => {
      expect(render()).toMatchSnapshot();
    });

    describe("Panel content", () => {
      describe("default", () => {
        const panels = render().root.findAllByType(SwitchSliderPanel);

        it('renders the text "OFF" in the panel', () => {
          expect(panels[0].props.children).toBe("OFF");
        });

        it("renders only one panel", () => {
          expect(panels.length).toBe(1);
        });
      });

      describe("when checked=true", () => {
        const panels = render({ checked: true }).root.findAllByType(
          SwitchSliderPanel
        );

        it('renders the text "ON" in the panel', () => {
          expect(panels[0].props.children).toBe("ON");
        });

        it("renders only one panel", () => {
          expect(panels.length).toBe(1);
        });
      });

      describe("when loading=true", () => {
        const panels = render({ loading: true }).root.findAllByType(
          SwitchSliderPanel
        );

        it("renders a Loader in the first panel", () => {
          expect(panels[0].props.children.type).toBe(Loader);
        });

        it("renders only one panel", () => {
          expect(panels.length).toBe(1);
        });
      });
    });

    describe("when checked=true", () => {
      it("applies the correct base styles", () => {
        const wrapper = render({ checked: true }).toJSON();
        assertStyleMatch(
          {
            backgroundColor: baseTheme.colors.primary,
          },
          wrapper
        );
      });

      it("applies the correct ::before styles", () => {
        const wrapper = render({ checked: true }).toJSON();
        assertStyleMatch(
          {
            transform: "translateX(36px)",
          },
          wrapper,
          { modifier: "::before" }
        );
      });
    });

    describe("when disabled=true", () => {
      it("applies the correct base styles", () => {
        const wrapper = render({ disabled: true }).toJSON();
        assertStyleMatch(
          {
            backgroundColor: baseTheme.disabled.background,
          },
          wrapper
        );
      });

      it("applies the correct ::before styles", () => {
        const wrapper = render({ disabled: true }).toJSON();
        assertStyleMatch(
          {
            opacity: "0.8",
          },
          wrapper,
          { modifier: "::before" }
        );
      });

      it("applies the correct SwitchSliderPanel styles", () => {
        const wrapper = render({ disabled: true }).toJSON();
        assertStyleMatch(
          {
            color: baseTheme.disabled.disabled,
          },
          wrapper,
          {
            modifier: css`
              ${SwitchSliderPanel}
            `,
          }
        );
      });
    });

    describe("when checked=true && disabled=true", () => {
      it("applies the correct SwitchSliderPanel styles", () => {
        const wrapper = render({ checked: true, disabled: true }).toJSON();
        assertStyleMatch(
          {
            color: baseTheme.colors.white,
          },
          wrapper,
          {
            modifier: css`
              ${SwitchSliderPanel}
            `,
          }
        );
      });
    });

    describe("when size=large", () => {
      describe("default", () => {
        it("applies the correct ::before styles", () => {
          const wrapper = render({ size: "large" }).toJSON();
          assertStyleMatch(
            {
              height: "36px",
              width: "36px",
            },
            wrapper,
            { modifier: "::before" }
          );
        });
      });

      describe("and checked=true", () => {
        it("applies the correct ::before styles", () => {
          const wrapper = render({ checked: true, size: "large" }).toJSON();
          assertStyleMatch(
            {
              transform: "translateX(38px)",
            },
            wrapper,
            { modifier: "::before" }
          );
        });
      });
    });
  });

  describe.each(carbonThemesJestTable)(
    "when the theme is set to %s",
    (themeName, theme) => {
      describe("default", () => {
        it("applies the correct base styles", () => {
          const wrapper = renderWithTheme({}, theme).toJSON();
          assertStyleMatch(
            {
              backgroundColor: theme.switch.off,
            },
            wrapper
          );
        });

        it("applies the correct ::before styles", () => {
          const wrapper = renderWithTheme({}, theme).toJSON();
          assertStyleMatch(
            {
              backgroundColor: theme.colors.white,
            },
            wrapper,
            { modifier: "::before" }
          );
        });
      });

      describe("and checked=true", () => {
        it("applies the correct base styles", () => {
          const wrapper = renderWithTheme({ checked: true }, theme).toJSON();
          assertStyleMatch(
            {
              backgroundColor: theme.colors.primary,
            },
            wrapper
          );
        });
      });

      describe("and disabled=true", () => {
        it("applies the correct base styles", () => {
          const wrapper = renderWithTheme({ disabled: true }, theme).toJSON();
          assertStyleMatch(
            {
              backgroundColor: theme.disabled.background,
            },
            wrapper
          );
        });

        it("applies the correct SwitchSliderPanel styles", () => {
          const wrapper = renderWithTheme({ disabled: true }, theme).toJSON();
          assertStyleMatch(
            {
              color: theme.disabled.disabled,
            },
            wrapper,
            {
              modifier: css`
                ${SwitchSliderPanel}
              `,
            }
          );
        });
      });

      describe("when checked=true && disabled=true", () => {
        it("applies the correct base styles", () => {
          const wrapper = renderWithTheme(
            { checked: true, disabled: true },
            theme
          ).toJSON();

          assertStyleMatch(
            {
              backgroundColor: theme.colors.disabled,
            },
            wrapper
          );
        });

        it("applies the correct SwitchSliderPanel styles", () => {
          const wrapper = renderWithTheme(
            { checked: true, disabled: true },
            theme
          ).toJSON();
          assertStyleMatch(
            {
              color: theme.colors.white,
            },
            wrapper,
            {
              modifier: css`
                ${SwitchSliderPanel}
              `,
            }
          );
        });
      });
    }
  );
});

function render(props) {
  return TestRenderer.create(<SwitchSlider {...props} />);
}

function renderWithTheme(props, theme, renderer = TestRenderer.create) {
  return renderer(
    <ThemeProvider theme={theme}>
      <SwitchSlider {...props} />
    </ThemeProvider>
  );
}
