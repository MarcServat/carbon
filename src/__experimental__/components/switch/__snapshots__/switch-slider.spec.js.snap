// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchSlider base theme renders as expected 1`] = `
.c1 {
  border: 0;
  color: #FFFFFF;
  margin: auto;
  margin-top: 5px;
}

.c1[type="on"] {
  margin-left: 9px;
}

.c1[type="off"] {
  color: rgba(0,0,0,0.9);
  margin-right: 6px;
}

.c0 {
  background-color: #CCD6DB;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 12px;
  font-weight: bold;
  height: 24px;
  left: 0;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  position: absolute;
  top: 0;
  width: 60px;
  z-index: 2;
}

.c0::before {
  background-color: #FFFFFF;
  bottom: 2px;
  box-shadow: 0 3px 3px 0 rgba(0,20,29,0.2),0 2px 4px 0 rgba(0,20,29,0.15);
  content: "";
  height: 20px;
  position: absolute;
  left: 2px;
  -webkit-transition: -webkit-transform 0.4s;
  -webkit-transition: transform 0.4s;
  transition: transform 0.4s;
  width: 20px;
  z-index: 1;
}

.c0 .sc-eCApnc {
  position: absolute;
  right: -30px;
  height: 100%;
}

<span
  className="c0"
>
  <div
    className="c1"
    type="off"
  >
    OFF
  </div>
</span>
`;
